### PR TITLE
rebuild-kernel: add flags to preserve container and use local gok-build-kernel

### PR DESCRIPTION
**Pull request by dev-zero**
I needed the build container to stay around to debug my build, and then I had local modifications to `cmd/gokr-build-kernel` I wanted to get picked up.

### CODE REVIEW

File: cmd/gokr-rebuild-kernel/kernel.go [hunk 1]
```diff
@@ -140,7 +146,7 @@ func main() {
 
 	buildPath := filepath.Join(tmp, "gokr-build-kernel")
 
-	cmd := exec.Command("go", "build", "-o", buildPath, "github.com/gokrazy/kernel/cmd/gokr-build-kernel")
+	cmd := exec.Command("go", "build", "-o", buildPath, *buildKernelProject+"/cmd/gokr-build-kernel")
```
Comment (stapelberg) [inline cmd/gokr-rebuild-kernel/kernel.go:149]: I don’t understand this change. The go build invocation only uses the import path to fully identify gokr-build-kernel, regardless of the current subdirectory. Modifications should already be picked up without this change, I thought.

**Review by stapelberg — CHANGES_REQUESTED** [review]
Thanks for the PR! The keep flag makes sense, the other one I left a question about.

File: cmd/gokr-rebuild-kernel/kernel.go (hunk 1 above)
Comment (dev-zero) [reply cmd/gokr-rebuild-kernel/kernel.go]: I believe you are right, and I mistook `build` for `get`. Thanks, removed.

**Comment by dev-zero** [comment]
as a side-question: would you accept a patch which adds automatic detection for an RPi kernel tree to the following part in case `gokr-build-kernel` is used on such a kernel tree?

https://github.com/gokrazy/kernel/blob/bc00f78631f893f6639e722dfcab990fd0f0294b/cmd/gokr-build-kernel/build.go#L188-L210

Something along the lines of:

```
if _, err := os.Stat("arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dtb""); err == nil {
	if err := copyFile("/tmp/buildresult/bcm2710-rpi-3-b.dtb", "arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dtb"); err != nil {
		log.Fatal(err)
	}
} else {
	if err := copyFile("/tmp/buildresult/bcm2710-rpi-3-b.dtb", "arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dtb"); err != nil {
		log.Fatal(err)
	}
}
```

The use case is that I need an RPi patched kernel tree for my pet project for soundcard (and sdxc) support, both patches are entangled with a lot of other things, including the device tree.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
> as a side-question: would you accept a patch which adds automatic detection for an RPi kernel tree to the following part in case `gokr-build-kernel` is used on such a kernel tree?

I had actually thought about going one step further and creating a repository for the rpi kernel. It seems like support for the Raspberry Pi 5 won’t land in the upstream linux kernel anytime soon unfortunately, so there’s only one available kernel for that platform :-/

**Pull request merged**